ApollosProject/apollos-apps
# Connected "Up Next" button

## DESCRIPTION

### What does this PR do, or why is it needed?

Companion PR: https://github.com/ApollosProject/apollos-templates/pull/20

This PR adds a connected component that will when rendered,

- Hide itself if it's rendered on a card with no children
- When rendered on a card with children, show a "continue" text when the card has uncompleted children
- When rendered on a card with children, all of which have been completed, render a disabled "finished" button 
- Addresses some complaints with content item ordering. https://github.com/ApollosProject/apollos-apps/issues/1247 

### What design trade-offs/decisions were made?

Things that need to be done
- Build in a "started" state. This will require API changes.

### How do I test this PR?

WIthout https://github.com/ApollosProject/apollos-apps/pull/1395 this will be hard to test without this component merged. My account is setup for testing, however, so if you reach out to me I can help you login to my apollos account. General steps look like this 

1. Pull this PR and companion PR. 
2. Login
3. Visit the discover tab.
4. Click on any content series icon
 - you should see the button load in, and render "continue". Pressing the button should take you to a piece of content
5. On The "Gauntlet 2018" content series you should see the "finished" state. 

![2020-03-25 15 35 11](https://user-images.githubusercontent.com/1637694/77579910-7508d000-6eb1-11ea-9119-b57209dbe3ca.gif)
![Simulator Screen Shot - iPhone 11 - 2020-03-25 at 15 34 49](https://user-images.githubusercontent.com/1637694/77579943-85b94600-6eb1-11ea-9171-a19f2ce218ee.png)



### What automated tests did you write?

## TODO:


## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
